scripts/ontology_to_ontology_mapping_tool/ontology_to_ontology_mapping_tool.py

```diff
@@ -0,0 +1,171 @@
+# Copyright (c) 2019 Memorial Sloan-Kettering Cancer Center.
+#
+# This library is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY, WITHOUT EVEN THE IMPLIED WARRANTY OF
+# MERCHANTABILITY OR FITNESS FOR A PARTICULAR PURPOSE.  The software and
+# documentation provided hereunder is on an "as is" basis, and
+# Memorial Sloan-Kettering Cancer Center
+# has no obligations to provide maintenance, support,
+# updates, enhancements or modifications.  In no event shall
+# Memorial Sloan-Kettering Cancer Center
+# be liable to any party for direct, indirect, special,
+# incidental or consequential damages, including lost profits, arising
+# out of the use of this software and its documentation, even if
+# Memorial Sloan-Kettering Cancer Center
+# has been advised of the possibility of such damage.
+
+import sys
+import os
+import io
+import argparse
+import pandas as pd
+import requests
+
+ONTOLOGY_MAPPINGS_FILE_URL = "https://raw.githubusercontent.com/cBioPortal/oncotree/master/scripts/ontology_to_ontology_mapping_tool/ontology_mappings.txt"
+VALID_ONCOTREE_CODES_URL = "https://raw.githubusercontent.com/cBioPortal/oncotree/master/resources/resource_uri_to_oncocode_mapping.txt"
+ACCEPTED_ONTOLOGIES = ['ONCOTREE_CODE','UMLS_CODE','NCIT_CODE','ICDO_CODE','HEMEONC_CODE']
+NULL_VALUES = ['','NA']
+
+def add_comments_column_and_log_data(mapped_data,target_file,source_code,target_code,source_file):
+    comments = []
+    completely_resolved_codes = {}
+    ambiguous_codes = {}
+    unrecognized_codes = []
+    many_to_one_codes = {}
+    mapped_codes_count = 0
+    unmapped_codes_count = 0
+
+    columns_to_groupby = mapped_data.columns[:len(mapped_data.columns)-1]
+    grouped_data = mapped_data.groupby(list(columns_to_groupby),sort=False)[target_code].unique().apply(', '.join).reset_index()
+    oncotree_codes_list = pd.read_csv(io.StringIO(requests.get(VALID_ONCOTREE_CODES_URL).content.decode('utf-8')),sep='\t',header=None,keep_default_na=False)
+    valid_oncotree_codes = oncotree_codes_list[2].str.upper().tolist()
+
+    for sc,tc in zip(grouped_data[source_code],grouped_data[target_code]):
```

Comment:
add space after comma in tuple lists too

Reply:
Done